lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -477,21 +494,110 @@ class MongoClientInterface {
         });
     }
 
+    _getPrevMD(c, params, isUpdate, log, cb) {
+        const collectionName = c.s.name;
+        const { objName, versionId } = params;
+        if (isUserBucket(collectionName)) {
+            if (isUpdate) {
+                return c.find({
```

Comment:
What is the complexity of this find operation? it looks like it perform a database request.

Reply:
It helps maintain counters accurate (because the update can have a different size than the version it's overwriting) and does a get on a version id. It's either this with full-scan every hour or full-scan way more often. So PUT/DELETE operations on versioned buckets go from 1 read + 1 write to 2 read + 1 write.